### PR TITLE
Civilian bounties and HAUL gauntlets

### PR DESCRIPTION
**Pull request by KoboldCommando**
## About The Pull Request
Ports cargo stuff from tg!  Including the H.A.U.L. Gauntlets, and the civilian bounty machines and system.

These things can be researched (data theory, and mining technology), and then printed at the cargo techfab.  The machines can be assembled and disassembled as normal.

Cargo bounties work just like tg.  You plug in your ID, grab your bounty, send it via the pad, and have cargo export the resulting bounty cube.  They can tag it with an export scanner for a tip!

Export scanners also no longer need to be linked to a console to work (it was pointless).

HAUL Gauntlets do not have an equip/unequip time.  I'm not sure what the reason for this was exactly and it was a pain to get working before Bee gets a deeper equip/unequip system rework.

An inserted ID can potentially wind up "stuck" in the computer if the pad gets disabled.  The computer has a line in the description reminding people that they can manually remove it with an alt-click.

The prices are not adjusted yet, and it pulls from the same list of bounties as the existing ones.  So it will likely need balance tweaks going forward.  But we can look into that once it's in place!

## Why It's Good For The Game

More cargo shit!  Also a system that encourages cargo to interact with the station more, and the gloves help that interaction by assuring crate safety.

Also a step toward a better economy!  After the early round, when early bounties have been knocked out and cargo starts to coast, they will now likely have access to the research to get this started, which gives cargo more to do and will pick up the ability for crew members to continue making money.

## Changelog
:cl:
add: Haul gauntlets can be researched and printed at cargo techfabs
add: Civilian bounty computers and pads can also be researched and printed at cargo
add: The civilian bounty system and bounty cubes is ported from tg.  Get to making that cash, spacemen!
tweak: export scanners no longer need to sync with a console
/:cl:

### CODE REVIEW

**Comment by WonderPsycho** [comment]
oooh me likey

File: code/__DEFINES/economy.dm [hunk 1]
```diff
@@ -26,3 +26,43 @@
 #define ACCOUNT_SEC_NAME "Defense Budget"
 
 #define NO_FREEBIES "commies go home"
+
+//Monkestation edit begin
+/// How much mail the Economy SS will create per minute, regardless of firing time.
+#define MAX_MAIL_PER_MINUTE 3
+/// Probability of using letters of envelope sprites on all letters.
+#define FULL_CRATE_LETTER_ODDS 70
```
Comment (nednaZ) [inline code/__DEFINES/economy.dm:34]: Mail stuff shouldn't be in your bounty port

File: code/__DEFINES/economy.dm [hunk 2]
```diff
@@ -26,3 +26,43 @@
...
+
+//Experimental change: These are subject to tweaking based on the /tg/ economy overhaul.
+//Current design direction: Higher paying jobs are vastly outnumbered by lower paying jobs, so anything above medium hurts inflation, common jobs help inflation
+#define PAYCHECK_PRISONER 25
```
Comment (nednaZ) [inline code/__DEFINES/economy.dm:38]: We don't have prisoners

**Comment by nednaZ** [comment]
What is the non-modular "icons/mob/hands.dmi" for in the changed files?

**Comment by KoboldCommando** [comment]
> What is the non-modular "icons/mob/hands.dmi" for in the changed files?

I had to add an on-mob sprite for the gloves.  The way they're written it's a battle to use any file other than hands.dmi

File: code/__DEFINES/economy.dm (hunk 1 above)
Comment (KoboldCommando) [reply code/__DEFINES/economy.dm:34]: I wound up copying the entire defines file from tg in a struggle to figure out what was missing, haha, I'll trim this up!

**Comment by nednaZ** [comment]
> > What is the non-modular "icons/mob/hands.dmi" for in the changed files?
> 
> I had to add an on-mob sprite for the gloves. The way they're written it's a battle to use any file other than hands.dmi

You should be able to use worn_icon for that. Take a look at the current state of the atmos hardsuits to see a fully modularized piece of clothing. (code/modules/clothing/spacesuits/hardsuit.dm)

**Comment by KoboldCommando** [comment]
hands.dmi modularized!

File: code/controllers/subsystem/economy.dm [hunk 3]
```diff
@@ -21,11 +21,32 @@ SUBSYSTEM_DEF(economy)
 	///Multiplied as they go to all department accounts rather than just cargo.
 	var/bounty_modifier = 3
 
-	//MonkeStation Edit: /tg/ Mail Port
+	//Monkestation edit begin
```
Comment (Archanial) [inline code/controllers/subsystem/economy.dm:24]: There is a pr on upstream porting tg mailing system, you might want to wait until it gets merged

Comment (nednaZ) [reply code/controllers/subsystem/economy.dm:24]: We actually already have it.
I ported it myself locally right before the upstream efforts started.

It wasn't a 1:1 port, so I didn't send it upstream

**Comment by nednaZ** [comment]
Requires fixes for pirate consoles before merge

**Comment by nednaZ** [comment]
Closing due to inactivity and merge conflicts.

**Pull request closed without merging**